File: From_Last_Year/Jutge/Week 8/Subwords_1.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
#include <set>

using namespace std;

/*
SUBWORDS(s)
	let S be an empty set (of strings)
	
	for i = 1 to |s| do
		for j = 1 to |s| do
			insert s[i,...,j] in S
	
	let L be an empty list (of strings)
	
	for all w in S do
		append w to L
	
	sort L
	return L
*/

vector <string> subwords(string s){

    set <string> S;

    for(int i = 0; i < s.size(); i++){
        for(int j = 0; j < s.size(); j++){
            S.insert(s.substr(i,j+1));
        }
    }

    vector <string> l;

    for (auto &w : S){
        l.push_back(w);
    }

    sort(l.begin(), l.end());

    return l;
}

int main(){
    string s;

    cin >> s;

    for(auto &x : subwords(s)){
        cout << x << endl;
    }
}

File: From_Last_Year/Jutge/Week 8/Subwords_1.py
import sys

"""
SUBWORDS(s)
	let S be an empty set (of strings)
	
	for i = 1 to |s| do
		for j = 1 to |s| do
			insert s[i,.......,j] in S
	
	let L be an empty list (of strings)
	
	for all w in S do
		append w to L
	
	sort L
	return L
	
"""
s = sys.stdin.readline().strip()

def subwords(s):
	S = set()
	for i in range(len(s)):
		for j in range(i,len(s)):
			S.add(s[i:j+1])
			print(S)
	
	l = []
	
	for x in S:
		l.append(x)
	
	l.sort()
	
	return l			

for x in subwords(s):
	print(x)
